ipatests/test_xmlrpc/test_schema_plugin.py

```diff
@@ -0,0 +1,143 @@
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""
+Test the `ipaserver/plugins/schema.py` module.
+"""
+
+import pytest
+
+from ipalib import errors
+from ipatests.test_xmlrpc.tracker.base import Tracker
+from ipatests.test_xmlrpc.xmlrpc_test import XMLRPC_test
+
+
+@pytest.mark.tier1
+class TestParamFindAndShowCommand(XMLRPC_test):
+    """Test functionality of the ipa param-{find,show} command"""
+    tracker = Tracker()
+
+    def test_param_find(self):
+        """Test param-find command"""
+
+        # right command without criteria
+        criteria = u'postalcode'
+        result = self.tracker.run_command('param_find', u'user-add')
+        assert len(result['result']) != 0, result
+
+        # right command, right criteria
+        criteria = u'postalcode'
+        result = self.tracker.run_command('param_find', u'user-add', criteria)
+        assert len(result['result']) != 0, result
+        for item in result['result']:
+            assert (criteria in item['name'].lower() or
+                    criteria in item['doc'].lower()), item
+
+        # wrong command, right criteria
+        result = self.tracker.run_command('param_find', u'fake', criteria)
+        assert len(result['result']) == 0, result
+
+        # right command, wrong criteria
+        result = self.tracker.run_command('param_find', u'user-add', u'fake')
+        assert len(result['result']) == 0, result
+
+        # wrong command, wrong criteria
+        result = self.tracker.run_command('param_find', u'fake', u'fake')
+        assert len(result['result']) == 0, result
+
+    def test_param_show(self):
+        """Test param-show command"""
+
+        # right command, right criteria
+        criteria = u'uid'
+        result = self.tracker.run_command('param_show', u'user-add', criteria)
+        assert result['result'] is not None, result
+        assert result['result']['name'] == 'uid', result
+
+        # right command without criteria
+        with pytest.raises(errors.RequirementError):
+            result = self.tracker.run_command('param_show', u'user-add')
+
+        # right command, wrong criteria
+        with pytest.raises(errors.NotFound):
+            result = self.tracker.run_command('param_show', u'user-add', u'fake')
+
+        # wrong command, right criteria
+        with pytest.raises(errors.NotFound):
+            result = self.tracker.run_command('param_show', u'fake', u'uid')
+
+        # wrong command, wrong criteria
+        with pytest.raises(errors.NotFound):
+            result = self.tracker.run_command('param_show', u'fake', u'fake')
+
+
+class TestOutputFindAndShowCommand(XMLRPC_test):
+    """Test functionality of the ipa output-{find,show} command"""
+    tracker = Tracker()
+
```

Comment:
Here is a philosophical question: is it worth comparing the output to something known or is it safe to assume that any output is ok? I'm not sure how much introspective is required and unless we want to test every single possible command's output it will still be hit or miss.